src/EnergyPlus/AirLoopHVACDOAS.cc

```diff
@@ -1023,33 +1023,36 @@ namespace AirLoopHVACDOAS {
         int constexpr winterDesignDayTypeIndex(10);
 
         for (size_t i = 1; i <= state.dataWeatherManager->DesDayInput.size(); i++) {
+            auto &desDayInput = state.dataWeatherManager->DesDayInput(i);
             // Summer design day
-            if (state.dataWeatherManager->DesDayInput(i).DayType == summerDesignDayTypeIndex && state.dataAirLoopHVACDOAS->SummerDesignDayFlag) {
-                this->SizingCoolOATemp = state.dataWeatherManager->DesDayInput(i).MaxDryBulb;
-                if (state.dataWeatherManager->DesDayInput(i).HumIndType == WeatherManager::DDHumIndType::WetBulb) { // wet bulb temperature
-                    this->SizingCoolOAHumRat = Psychrometrics::PsyWFnTdbTwbPb(
-                        state, this->SizingCoolOATemp, state.dataWeatherManager->DesDayInput(i).HumIndValue, DataEnvironment::StdPressureSeaLevel);
-                } else if (state.dataWeatherManager->DesDayInput(i).HumIndType == WeatherManager::DDHumIndType::DewPoint) { // dewpoint
-                    this->SizingCoolOAHumRat = Psychrometrics::PsyWFnTdpPb(
-                        state, state.dataWeatherManager->DesDayInput(i).HumIndValue, DataEnvironment::StdPressureSeaLevel);
-                } else if (state.dataWeatherManager->DesDayInput(i).HumIndType == WeatherManager::DDHumIndType::HumRatio) {
-                    this->SizingCoolOAHumRat = state.dataWeatherManager->DesDayInput(i).HumIndValue;
-                } // else { // What about other cases?
-                state.dataAirLoopHVACDOAS->SummerDesignDayFlag = false;
+            if (desDayInput.DayType == summerDesignDayTypeIndex) {
+                // keep design day humiditiy ratio from the same peak temperature design day (i.e., don't use a humrat from a different day)
+                if (desDayInput.MaxDryBulb > this->SizingCoolOATemp) {
```

Comment:
The new methodology is to search through all design days and select the peak condition and save both T and w to use for sizing. Because this function is only called once for sizing there is no need for the Summer/WinterDesignDayFlag. Also, T and w are always from the same design day, using the peak outdoor T as the indicator of peak and then saving w at the same time (i.e., T and w are not evaluated seperately).